### PR TITLE
ComboBox: Respect ariaLabel from IComboBoxOption when setting aria-label on item

### LINKED ISSUE #6665: ComboBox: aria label of an item cannot be set separately from preview text
### Bug Report 

- __Package version(s)__: 6.74.0
 
#### Priorities and help requested:
 
Are you willing to submit a PR to fix? No
 
Requested priority: High (accessibility)
 
Products/sites affected: Outlook
 
#### Describe the issue:
 
Setting the ariaLabel property on IComboBoxOption does not set the aria-label on the option. It only does so if useAriaLabelAsText is set to true, which results in the aria label being used as the preview text.
 
Related PRs:
https://github.com/OfficeDev/office-ui-fabric-react/pull/4540
https://github.com/OfficeDev/office-ui-fabric-react/pull/4722

#### Actual behavior:
 
Aria label can only be set if you want to use it as the preview text as well
 
#### Expected behavior:
 
Aria label can be set, and if you want to use it as the preview text you can use the existing useAriaLabelAsText property

See ComboBox._renderOption and ComboBox._getPreviewText, essentially leave _getPreviewText as is but update the aria-label props being set on the CommandButton or CheckBox to be "item.ariaLabel ? item.ariaLabel : item.text"

### PR DESCRIPTION
**Pull request by janormanms**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #6665 
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Setting the ariaLabel property on IComboBoxOption does not set the aria-label on the option. It only does so if useAriaLabelAsText is set to true, which results in the aria label being used as the preview text

Leaving the useAriaLabelAsText behavior the same for the preview text case, but using ariaLabel if we have it for the aria-label

#### Focus areas to test

ComboBox accessibility

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx [hunk 1]
```diff
@@ -1175,7 +1176,7 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
           onMouseLeave={this._onOptionMouseLeave}
           role="option"
           aria-selected={isSelected ? 'true' : 'false'}
-          ariaLabel={this._getPreviewText(item)}
+          ariaLabel={ariaLabel}
```
Comment (jspurlin) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: The `getPreviewText` added for some specific cases and the but this PR is trying to address shouldn't be fixed like this.

**Comment by janormanms** [comment]
The use case for this scenario is an end time picker which displays the time as the title/selected value of the combobox, but the time plus duration for each option in the dropdown:
image

Since the style of the time (the option text) is different than that of the duration these are two different strings and it is currently not possible to have Narrator read the time and duration when going through the items in the dropdown without also rendering the duration (without styling) as the preview text.

**Comment by janormanms** [comment]
I do want the screen readers to read the time and the duration while the user is navigating in the dropdown, that was my intention with this change.

Are you saying that if I throw a (unique) id on the span displaying the duration for each option (which we render during onRenderOption) and then set ariaDescribedBy to that id for each option that it would have the same effect?

I'll give that a try this morning to see if that or a similar fix can solve this issue without a code change here.

As a consumer of this control though I would still find it odd that the props allow for setting an ariaLabel but that it is only applied if we also want it to be used for the preview text.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by jspurlin** [comment]
Yeah, I agree that it's not the most straightforward, but the current behavior helps make sure that a screen reader user will be able to type values in the input that correspond values that they will hear.

An alternative solution would be to introduce some like a `previewText` prop and at that point you could pass in `text='8:30 PM (1.5 hours)'` and `previewText='8:30 PM'`. At that point the user would hear the full string but only 8:30 PM would be used in the input (both visually and when attempting to find matches) without the need to use an ariaLabel. The downside of that is that now the option text does not match what the user hears/sees (which I know is what you want, but I don't know if I'm a huge fan of that behavior)

**Pull request closed without merging**